grakn/service/Session/TransactionService.py

```diff
@@ -39,16 +40,11 @@ def __init__(self, session_id, tx_type, transaction_endpoint):
         self._communicator.send(open_req)
     __init__.__annotations__ = {'tx_type': enums.TxType}
 
-
     # --- Passthrough targets ---
     # targets of top level Transaction class
 
     def query(self, query, infer=True):
-        request = RequestBuilder.query(query, infer=infer)
-        # print("Query request: {0}".format(request))
-        response = self._communicator.send(request)
-        # convert `response` into a python iterator
-        return ResponseReader.ResponseReader.query(self, response.query_iter)
+        return imap(ResponseReader.ResponseReader.query(self), Iterator(self._communicator, RequestBuilder.start_iterating_query(query, infer)))
```

Comment:
can you write a comment on what this imap does here?

Reply:
It's mapping the response reader to each result of the iterator, is the comment definitely necessary?

Reply:
It's just hard to read what is intended to be going on here: are we mappingthe query() function with new arguments generated by the iterator? Does that mean the `query(self)` method returns a function that takes further arguments?

Functional style programming is great but i find it hard to follow when mixed in with other styles :)

Reply:
Moved the `imap` to `ResponseReader` to be consistent with the style for `get_attributes_by_value`. This method previously existed and was called `query` but returned an iterator that it created, since iteration was handled by `ResponseReader` prior. I have also renamed the method to `get_query_results` now, and it takes an iterator as as parameter rather than being mapped (so that the `imap` and the `lambda` are in the same place).

Reply:
i think if you rename `query(self)` above like i mentioned in other comment it will help a lot, right now this just isn't readable standalone :)